packages/contracts-ecosystem/script/DeployFaucet.s.sol

```diff
@@ -0,0 +1,308 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {Script} from 'forge-std/Script.sol';
+import {console} from 'forge-std/console.sol';
+
+import {ProxyAdmin} from '@eth-optimism/contracts-bedrock/src/universal/ProxyAdmin.sol';
+import {Proxy} from '@eth-optimism/contracts-bedrock/src/universal/Proxy.sol';
+import {Faucet} from '@eth-optimism/contracts-bedrock/src/periphery/faucet/Faucet.sol';
+import {AdminFaucetAuthModule} from '@eth-optimism/contracts-bedrock/src/periphery/faucet/authmodules/AdminFaucetAuthModule.sol';
+
+contract DeployFaucet is Script {
+  uint256 deployerPrivateKey;
+  uint256 proxyAdminOwnerPrivateKey;
+  uint256 faucetAdminPrivateKey;
+  uint256 faucetOnchainAuthModuleTtl;
+  uint256 faucetOnchainAuthModuleAmount;
+  uint256 faucetOffchainAuthModuleTtl;
+  uint256 faucetOffchainAuthModuleAmount;
+
+  address faucetAdmin;
+  address proxyAdminOwner;
+  address proxyAdminContract;
+  address faucetProxyContract;
+  address faucetContract;
+  address faucetOnchainAuthModuleAdmin;
+  address onChainAuthModuleContract;
+  address faucetOffchainAuthModuleAdmin;
+  address offChainAuthModuleContract;
+
+  function setUp() public {
+    deployerPrivateKey = vm.envUint('DEPLOYER_PRIVATE_KEY');
+
+    proxyAdminOwnerPrivateKey = vm.envUint('PROXY_ADMIN_OWNER_PRIVATE_KEY');
+    proxyAdminOwner = vm.createWallet(proxyAdminOwnerPrivateKey).addr;
+
+    faucetOnchainAuthModuleAmount = vm.envUint(
```

Comment:
Noticed we moved out from config file to env vars. Maybe we want to store the latest deployment values somewhere like 1pass so its easy for anyone to rerun?

Reply:
agreed!